### PR TITLE
datepicker fixed

### PR DESCRIPTION
**Pull request by carloszv**

### CODE REVIEW

File: src/components/DatePicker.tsx [hunk 1]
```diff
@@ -86,6 +92,26 @@ export type Props = {
...
+
+function formatDate(date: Date): string {
+  const formattedDate = `${getValidYearValue(
+    `${date.getFullYear()}`,
+  )}-${getValidDateOrMonthValue(
+    `${date.getMonth() + 1}`,
+  )}-${getValidDateOrMonthValue(`${date.getDate()}`)}`
+
+  return formattedDate
+}
```
Comment (lukezirngibl) [inline src/components/DatePicker.tsx:113]: can we use date-fns format instead?

File: src/components/DatePicker.tsx [hunk 2]
```diff
@@ -86,6 +92,26 @@ export type Props = {
...
 
+function getValidDateOrMonthValue(value: string): string {
+  const resultValue = value.length > 1 ? value : `0${value}`
+  return resultValue
+}
+
+function getValidYearValue(value: string): string {
+  const resultValue = value.length == 4 ? value : `1900`
+  return resultValue
+}
```
Comment (lukezirngibl) [inline src/components/DatePicker.tsx:103]: I think we'll want a more general solution than this

File: src/components/DatePicker.tsx [hunk 3]
```diff
@@ -145,47 +183,36 @@ export const DatePicker = (props: Props) => {
...
               dataTestId={props.dataTestId}
             />
           ) : (
             <>
               <TextInput
                 hasFocus={props.hasFocus}
                 onChange={() => {}}
+                maxLength={4}
+                pattern={'[1-9][0-9]{3}'}
+                max={'9999-12-31'}
```
Comment (lukezirngibl) [inline src/components/DatePicker.tsx:196]: what does this do?

File: src/components/DatePicker.tsx [hunk 4]
```diff
@@ -145,47 +183,36 @@ export const DatePicker = (props: Props) => {
...
-              value={props.value ? format(props.value, props.dateFormat) : null}
+              value={props.value ? textInputValue : null}
               dataTestId={props.dataTestId}
             />
           ) : (
             <>
               <TextInput
                 hasFocus={props.hasFocus}
                 onChange={() => {}}
+                maxLength={4}
```
Comment (lukezirngibl) [inline src/components/DatePicker.tsx:194]: shouldn't max length be more than 4?

File: src/components/DatePicker.tsx (hunk 3 above)
Comment (carloszv) [reply src/components/DatePicker.tsx:196]: this disables the possibility to ty a year like 123456

**Pull request closed without merging**